includes/MediaWiki/Specials/SpecialWikibaseReconcileEdit.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Specials;
+
+use HTMLForm;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request\EditRequestParser;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\Request\EditRequestSaver;
+use MediaWiki\Extension\WikibaseReconcileEdit\MediaWiki\WikibaseReconcileEditServices;
+use MediaWiki\Rest\LocalizedHttpException;
+use SpecialPage;
+use Status;
+use Wikibase\DataModel\Entity\PropertyId;
+use Wikibase\Lib\Store\EntityRevision;
+use Wikibase\Repo\WikibaseRepo;
+
+/**
+ * @license GPL-2.0-or-later
+ */
+class SpecialWikibaseReconcileEdit extends SpecialPage {
+
+	/** @var string */
+	private $reconcileDefaultJSON;
+
+	/** @var string */
+	private	$entityDefaultJSON;
+
+	/** @var EditRequestParser */
+	private $editRequestParser;
+
+	/** @var EditRequestSaver */
+	private $editRequestSaver;
+
+	public function __construct() {
+		parent::__construct( 'WikibaseReconcileEdit' );
+
+		$this->editRequestParser = WikibaseReconcileEditServices::getEditRequestParser();
+		$this->editRequestSaver = WikibaseReconcileEditServices::getEditRequestSaver();
+
+		$propertyId = new PropertyId( 'P1' );
+
+		$this->reconcileDefaultJSON = json_encode( [
+			EditRequestParser::VERSION_KEY => '0.0.1',
+			'urlReconcile' => $propertyId->serialize(),
+		], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES );
+
+		$this->entityDefaultJSON = json_encode( [
+			EditRequestParser::VERSION_KEY => '0.0.1/minimal',
+			'statements' => [
+				[
+					'property' => $propertyId->serialize(),
+					'value' => "http://example.com/",
+				],
+			],
+		], JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES );
+	}
+
+	public static function factory(): self {
+		return new self();
+	}
+
+	public function getDescription(): string {
+		return $this->msg( 'wikibasereconcileedit-special-description' )->text();
+	}
+
+	protected function getGroupName(): string {
+		return 'wikibase';
+	}
+
+	public function execute( $subPage ) {
+		parent::execute( $subPage );
+		$this->requireLogin();
+		$this->showRequestForm();
+
+		$out = $this->getOutput();
+		$out->addModules( 'ext.WikibaseReconcileEdit' );
+	}
+
+	public function sendRequest( $data ) {
```

Comment:
I suggest renaming this, since I wouldn’t say it sends a request now. Looking at [`setSubmitCallback` codesearch results](https://codesearch.wmcloud.org/search/?q=setSubmitCallback%5C(%20%5C%5B%20%5C%24this&i=nope&files=&excludeFiles=&repos=), `onSubmit` or `submitForm` seem to be popular choices, and I think either of those would be fine – it should be okay to not mention in the method name _what_ the submit does, since there’s only one submit method anyways.

Reply:
okey, done